File: go/problem38.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	fmt.Println(LargestPandigitalMultiples())
}

func LargestPandigitalMultiples() int {

	maxPandigitNum := 1
	for i := 1; i < 20000; i++ {
		temp := ""
		for j := 1; j < 10; j++ {
			prod := i * j
			temp = temp + strconv.Itoa(prod)
			if len(temp) < 9 {
				continue
			} else if len(temp) == 9 {

				if isPandigital(temp) {
					val, err := strconv.Atoi(temp)
					if err != nil {
						fmt.Println("Result will be incorrect, error in converting string to int ", err)
						continue
					}
					if val > maxPandigitNum {
						maxPandigitNum = val
					}
				}

			} else {
				break
			}

		}
	}
	return maxPandigitNum
}

func isPandigital(numAsStr string) bool {

	digits := []string{"1", "2", "3", "4", "5", "6", "7", "8", "9"}
	for i := 0; i < len(digits); i++ {
		if strings.Index(numAsStr, digits[i]) == -1 {
			return false
		}
	}
	return true
}

File: python/problem38.py
def largestPandigitalMultiples():
    maxPandigitNum = 1
    for i in range(1, 20000):
        temp = ""
        for j in range(1, 10):
            prod = i * j
            temp = temp + str(prod)
            if len(temp) < 9:
                continue
            elif len(temp) == 9:
                if isPandigital(temp) and int(temp) > maxPandigitNum:
                        maxPandigitNum = int(temp)
            else:
                break

    return maxPandigitNum


def isPandigital(numAsStr):
    digits = ["1", "2", "3", "4", "5", "6", "7", "8", "9"]
    for i in range(0, len(digits)):
        if numAsStr.find(digits[i]) == -1:
            return False

    return True


if __name__=="__main__":
    print(largestPandigitalMultiples())
